test/e2e/framework/test_context.go

```diff
@@ -349,24 +349,6 @@ func RegisterClusterFlags(flags *flag.FlagSet) {
 	flags.DurationVar(&nodeKiller.SimulatedDowntime, "node-killer-simulated-downtime", 10*time.Minute, "A delay between node death and recreation")
 }
 
-// RegisterNodeFlags registers flags specific to the node e2e test suite.
-func RegisterNodeFlags(flags *flag.FlagSet) {
-	// Mark the test as node e2e when node flags are api.Registry.
-	TestContext.NodeE2E = true
-	flags.StringVar(&TestContext.NodeName, "node-name", "", "Name of the node to run tests on.")
```

Comment:
This NodeName is used in e2e/framework/pods.go except e2e_node, so I'd like to take more time for digging.

Reply:
NodeName is only used if the test is e2e_node like
e2e/framework/pods.go:
```
181 // mungeSpec apply test-suite specific transformations to the pod spec.
182 func (c *PodClient) mungeSpec(pod *v1.Pod) {
183         if !TestContext.NodeE2E {
184                 return
185         }
186 
187         gomega.Expect(pod.Spec.NodeName).To(gomega.Or(gomega.BeZero(), gomega.Equal(TestContext.NodeName)), "Test misconfigured")
188         pod.Spec.NodeName = TestContext.NodeName
```
So it is fine to move this flag to e2e_node test.